feat(ErrorHandler.class, ErrorResponse.class): добавить централизован…

CheTigor opened the pull request:
…ную обработку исключений

Теперь при возникновении каких-либо исключений будет передаваться правильный код и понятный ответ пользователю

refactor(UserStorage.class, FilmStorage.class): выделить классы для хранения информации

Добавлены новые классы, для хранения и модифицирования данных

feat(UserService.class, FilmService.class): добавить классы с бизнес-логикой

Добавлен слой сервисов

feat(FilmController.class, UserController.class): добавить новые запросы

Новые запросы для User:
- getUserById
- addFriend
- removeFriend
- getUserFriends
- getCommonFriends Новые запросы для Film:
- getFilmById
- addLike
- removeLike
- getPopularFilms

CheTigor committed:
feat(ErrorHandler.class, ErrorResponse.class): добавить централизованную обработку исключений

Теперь при возникновении каких-либо исключений будет передаваться правильный код и понятный ответ пользователю

refactor(UserStorage.class, FilmStorage.class): выделить классы для хранения информации

Добавлены новые классы, для хранения и модифицирования данных

feat(UserService.class, FilmService.class): добавить классы с бизнес-логикой

Добавлен слой сервисов

feat(FilmController.class, UserController.class): добавить новые запросы

Новые запросы для User:
- getUserById
- addFriend
- removeFriend
- getUserFriends
- getCommonFriends
Новые запросы для Film:
- getFilmById
- addLike
- removeLike
- getPopularFilms

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,74 @@
+            throw new LikeNotFoundException(String.format(
+                    "Пользователь с id: %d не ставил лайк фильму с id: %d", userId, filmId));
+        }
+        film.getUserLikes().remove(userId);
```
удаление через storage
это же контракт. 
Завтра будет дургая реализация хранилища и ты уже вот так film.getUserLikes().remove(userId);
удалить не сможешь. 
А если бы вызывал метод для удаления. То для сервиса бы ничего ен поменялось, по прежнемы бы просто вызывал метод удаления. 

Тоже самое касается юзера

CheTigor committed:
style: единообразие для параметров метода
  Добавлен перенос только когда это необходимо
fix(User, Film): добавить аннотацию JsonIgnore для полей объектов
  Для полей userLikes, friendsId добавлены новые аннотации
fix(InMemoryUserStorage, InMemoryFilmStorage): изменена логика присвоения id
  Теперь метод использует новое значение id, а не увеличивает его для следующего
fix(InMemoryUserStorage, InMemoryFilmStorage): сделать класс согласно CRUD
  Методы создания, удаления, обновления и чтения объектов из хранилища перемещены из класса сервиса в класс хранения

CheTigor committed:
style(README.md): написано описание
Добавлена ER диаграмма
